lib/galaxy/objectstore/cloud.py

```diff
@@ -0,0 +1,615 @@
+"""
+Object Store plugin for Cloud storage.
+"""
+
+import logging
+import multiprocessing
+import os
+import shutil
+import subprocess
+import threading
+import time
+
+from datetime import datetime
+
+from galaxy.exceptions import ObjectInvalid, ObjectNotFound
+from galaxy.util import (
+    directory_hash_id,
+    safe_relpath,
+    string_as_bool,
+    umask_fix_perms,
+)
+from galaxy.util.sleeper import Sleeper
+
+from ..objectstore import convert_bytes, ObjectStore
+from cloudbridge.cloud.factory import CloudProviderFactory, ProviderList
+
+try:
+    # Imports are done this way to allow objectstore code to be used outside of Galaxy.
+    import boto
+
+    from boto.exception import S3ResponseError
+    from boto.s3.key import Key
```

Comment:
@mvdbeek Thanks, the imports are updated accordingly. 
Actually, `boto` import is temporary solution for handling AWS S3 exceptions until `cloudbridge` implements such exceptions for all the providers.